File: programs in csharp/simpify-path.cs
public class Solution {
    public string SimplifyPath(string path) {
        string[] pathArray = path.Split('/');

         Stack<string> stack = new Stack<string>();

        foreach(string c in pathArray){
            if(string.IsNullOrEmpty(c) || c == "."){
                continue;
            }else if(c == "..") {
                if(stack.Count > 0){
                    stack.Pop();
                }
            }else{
                stack.Push(c);
            }
        }

        string simplePath = "/" + string.Join("/", stack.Reverse());
        return simplePath;
    }
}

File: programs in python/simpify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = [] 
        pathArray = path.split("/")

        for c in pathArray:
            if c == "" or c == ".":
                continue
            elif c == "..":
                if stack:
                    stack.pop()
            else:
                stack.append(c)
            
        return "/" + "/".join(stack)